libraries/appbase/application.cpp

```diff
@@ -541,6 +540,14 @@ application& application_factory::new_application( const string& id){
    return apps.at(id);
 }
 
+
+application& application_factory::new_application( const string& id, application app){
+   if(apps.count(id))
+      apps.erase(id);
+   apps.emplace(id, id);
+   return apps.at(id);
+}
```

Comment:
what is app argument good for here ? You could also return apps.emplace(id, id)...

Reply:
std::map::emplace returns pair<iterator, bool>... so it will still require `return *(apps.emplace(id,id).first);`. I consider the first option more readable.

Reply:
ok, might be better to prefer readability compared to the efficiency(no need to call at...)

Reply:
In this case, as it is called only once for each chain created, the efficiency can be traded.